### PR TITLE
[vcpkg] [cudnn] [msmpi] [openmpi] Update VMSS

### LINKED ISSUE #11451: CI should be updated to Visual Studio 2019 16.6 released 2020-05-19
See:
https://devblogs.microsoft.com/visualstudio/visual-studio-2019-v16-6-and-v16-7-preview-1-ship-today/

### PR DESCRIPTION
**Pull request by BillyONeal**
This update is primarily to take fixes pushed by the Azure DevOps team for the Linux agents. Additionally, parts of the following VM update requests have been rolled up into this PR:

* https://github.com/microsoft/vcpkg/pull/11321
* https://github.com/microsoft/vcpkg/pull/11378
* https://github.com/microsoft/vcpkg/pull/11373
* https://github.com/microsoft/vcpkg/pull/11596

Special thanks to @JackBoosY for fixing a bunch of ports detected by this change.

Resolves #11451

### CODE REVIEW

File: scripts/azure-pipelines/windows/provision-image.ps1 [hunk 1]
```diff
@@ -112,14 +112,14 @@ $Workloads = @(
...
-
-$CudaUrl = 'https://developer.download.nvidia.com/compute/cuda/10.1/Prod/local_installers/cuda_10.1.243_426.00_win10.exe'
-$CudaFeatures = 'nvcc_10.1 cuobjdump_10.1 nvprune_10.1 cupti_10.1 gpu_library_advisor_10.1 memcheck_10.1 ' + `
-  'nvdisasm_10.1 nvprof_10.1 visual_profiler_10.1 visual_studio_integration_10.1 cublas_10.1 cublas_dev_10.1 ' + `
-  'cudart_10.1 cufft_10.1 cufft_dev_10.1 curand_10.1 curand_dev_10.1 cusolver_10.1 cusolver_dev_10.1 cusparse_10.1 ' + `
-  'cusparse_dev_10.1 nvgraph_10.1 nvgraph_dev_10.1 npp_10.1 npp_dev_10.1 nvrtc_10.1 nvrtc_dev_10.1 nvml_dev_10.1 ' + `
-  'occupancy_calculator_10.1 fortran_examples_10.1'
+$MpiUrl = 'https://download.microsoft.com/download/a/5/2/a5207ca5-1203-491a-8fb8-906fd68ae623/msmpisetup.exe'
+
+$CudaUrl = 'http://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda_10.2.89_441.22_win10.exe'
```
Comment (BillyONeal) [inline scripts/azure-pipelines/windows/provision-image.ps1:117]: Hmmm this should have been left as HTTPS

Comment (cenit) [reply scripts/azure-pipelines/windows/provision-image.ps1:117]: You are right sorry. I copy/pasted from browser without noticing

**Comment by BillyONeal** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/11494

**Comment by BillyONeal** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/11570

**Comment by BillyONeal** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/11571

**Comment by JackBoosY** [comment]
That's hard work. Thanks @BillyONeal !

**Comment by Voskrese** [comment]
SDK new ver msvc new sdk ?



Microsoft.VisualStudio.Component. **Windows10SDK.19041**

https://github.com/jpark37/virtual-environments/commit/8b4b249baa55ab2142d3e9ca89383262cccca961

**Comment by Voskrese** [comment]
```
-- Downloading https://sourceforge.net/projects/aifftools/files/libaiff/LibAiff%205.0/libaiff-5.0-release.tar.gz... Failed. Status: 7;"Couldn't connect to server"
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:175 (message):
      
      Failed to download file.
      If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
      variables to "https://user:password@your-proxy-ip-address:port/".
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues
```

```
-- Downloading https://sourceforge.net/projects/libusb-win32/files/libusb-win32-releases/1.2.6.0/libusb-win32-src-1.2.6.0.zip/download... Failed. Status: 7;"Couldn't connect to server"
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:175 (message):
      
      Failed to download file.
      If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
      variables to "https://user:password@your-proxy-ip-address:port/".
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues
```

re-run ???

**Review by JackBoosY — APPROVED** [review]

File: scripts/azure-pipelines/azure-pipelines.yml [hunk 2]
```diff
@@ -2,52 +2,88 @@
...
-      poolName: $(windows-pool)
-
-  - template: osx/azure-pipelines.yml
-
-  - template: linux/azure-pipelines.yml
-    parameters:
-      poolName: $(linux-pool)
+  windows-pool: 'PrWin-2020-06-04'
+  linux-pool: 'PrLin-2020-06-08'
+  delete-tombstones-first: false
```
Comment (strega-nil) [inline scripts/azure-pipelines/azure-pipelines.yml:7]: How does one set this variable?

**Review by strega-nil — APPROVED** [review]

File: scripts/azure-pipelines/azure-pipelines.yml (hunk 2 above)
Comment (BillyONeal) [reply scripts/azure-pipelines/azure-pipelines.yml:7]: You can make a PR that turns it on then turns it back off, or you can set it with the web interface.

**Pull request merged**